uportal-war/src/main/java/org/jasig/portal/utils/LdapQueryUtils.java

```diff
@@ -0,0 +1,34 @@
+package org.jasig.portal.utils;
+
+public class LdapQueryUtils {
+
+	/**
+	 * 
+	 * @param query
+	 * @return
+	 */
+	public static String escapeSpecialCharacters(String query) {
+        // We need to escape regex special characters that appear in the query string...
+        final String[][] specials = new String[][] {
+                            /* backslash must come first! */
+                            new String[] { "\\", "\\\\"}, 
+                            new String[] { "[", "\\[" }, 
+                            /* closing ']' isn't needed b/c it's a normal character w/o a preceding '[' */
+                            new String[] { "{", "\\{" }, 
+                            /* closing '}' isn't needed b/c it's a normal character w/o a preceding '{' */
+                            new String[] { "^", "\\^" },
+                            new String[] { "$", "\\$" },
+                            new String[] { ".", "\\." },
+                            new String[] { "|", "\\|" },
+                            new String[] { "?", "\\?" },
+                            new String[] { "*", "\\*" },
+                            new String[] { "+", "\\+" },
+                            new String[] { "(", "\\(" },
+                            new String[] { ")", "\\)" }
+                        };
```

Comment:
I think the above is a bit overkill.  Per section 4 of https://docs.ldap.com/specs/rfc4515.txt:
>   The <valueencoding> rule ensures that the entire filter string is a
>   valid UTF-8 string and provides that the octets that represent the
>   ASCII characters "*" (ASCII 0x2a), "(" (ASCII 0x28), ")" (ASCII
>   0x29), "\" (ASCII 0x5c), and NUL (ASCII 0x00) are represented as a
>   backslash "\" (ASCII 0x5c) followed by the two hexadecimal digits
>   representing the value of the encoded octet.

I believe only those characters need to be escaped, and the preceeding backslash AFAIK is an acceptable escaping mechanism.  The other characters are for RegEx-related comparisons which does not apply for an LDAP search filter term.  I don't think they would hurt, but it does add confusion IMHO so I'd remove the extraneous substitutions.

The above looks like a copy from https://github.com/Jasig/uPortal/blob/master/uportal-war/src/main/java/org/jasig/portal/groups/smartldap/SmartLdapGroupStore.java#L361.  The use cases are different.  In that case the keys (LDAP attribute names) are being compared using a regex expresssion so a regex substitution is reasonable.  However for LDAPGroupStore you are escaping the actual LDAP Search Filter.